--- a/Makefile
+++ b/Makefile
@@ -2,11 +2,12 @@
 # GNU make is required.
 
 # Your C compiler
-CC ?= gcc
+#CC ?= gcc
 #CC ?= clang
 
 # Enable compiler warnings. This is an absolute minimum.
 CFLAGS += -Wall -std=c99 -O3 #-Wextra
+CFLAGS += $(shell dpkg-buildflags --get CFLAGS)
 
 # Define your optimization flags.
 #
@@ -18,6 +19,9 @@
 # Define where you want Frotz installed
 PREFIX ?= /usr/local
 MANDIR ?= $(PREFIX)/share/man
+#BINDIR ?= $(PREFIX)/bin
+BINDIR ?= $(PREFIX)/games
+MAN_PREFIX ?= $(PREFIX)/share
 SYSCONFDIR ?= /etc
 
 # Choose your sound support
@@ -260,7 +264,7 @@
 curses: $(FROTZ_BIN)
 ncurses: $(FROTZ_BIN)
 $(FROTZ_BIN): $(FROTZ_LIBS)
-	$(CC) $+ -o $@$(EXTENSION) $(LDFLAGS) $(CURSES_LDFLAGS) $(CURSES_SOUND_LDFLAGS)
+	$(CC) $+ -o $@$(EXTENSION) $(LDFLAGS) $(CURSES_LDFLAGS) $(CURSES_SOUND_LDFLAGS) $(shell dpkg-buildflags --get LDFLAGS)
 	@echo "** Done building Frotz with curses interface"
 	@echo "** Audio support $(CURSES_SOUND) (type $(SOUND_TYPE))"
 	@echo "** Blorb support $(BLORB_SUPPORT)"
@@ -274,13 +278,13 @@
 
 dumb: $(DFROTZ_BIN)
 $(DFROTZ_BIN): $(DFROTZ_LIBS)
-	$(CC) $+ -o $@$(EXTENSION) $(LDFLAGS)
+	$(CC) $+ -o $@$(EXTENSION) $(LDFLAGS) $(shell dpkg-buildflags --get LDFLAGS)
 	@echo "** Done building Frotz with dumb interface."
 	@echo "** Blorb support $(BLORB_SUPPORT)"
 
 sdl: $(SFROTZ_BIN)
 $(SFROTZ_BIN): $(SFROTZ_LIBS)
-	$(CC) $+ -o $@$(EXTENSION) $(LDFLAGS) $(SDL_LDFLAGS)
+	$(CC) $+ -o $@$(EXTENSION) $(LDFLAGS) $(SDL_LDFLAGS) $(shell dpkg-buildflags --get LDFLAGS)
 	@echo "** Done building Frotz with SDL interface."
 
 dos: $(DOS_BIN)
@@ -439,36 +443,36 @@
 #
 install: install_frotz
 install_frotz: $(FROTZ_BIN)
-	install -d "$(DESTDIR)$(PREFIX)/bin" "$(DESTDIR)$(MANDIR)/man6"
-	install "frotz$(EXTENSION)" "$(DESTDIR)$(PREFIX)/bin/"
-	install -m 644 doc/frotz.6 "$(DESTDIR)$(MANDIR)/man6/"
+	install -d $(BINDIR) $(MAN_PREFIX)/man/man6
+	install -c -m 755 $(FROTZ_BIN) $(BINDIR)
+	install -m 644 doc/frotz.6 $(MAN_PREFIX)/man//man6/
 
 uninstall: uninstall_frotz
 uninstall_frotz:
-	rm -f "$(DESTDIR)$(PREFIX)/bin/frotz"
-	rm -f "$(DESTDIR)$(MANDIR)/man6/frotz.6"
+	rm -f $(BINDIR)/frotz
+	rm -f $(MAN_PREFIX)/man6/frotz.6
 
 install_dumb: install_dfrotz
 install_dfrotz: $(DFROTZ_BIN)
-	install -d "$(DESTDIR)$(PREFIX)/bin" "$(DESTDIR)$(MANDIR)/man6"
-	install "$(DFROTZ_BIN)" "$(DESTDIR)$(PREFIX)/bin/"
-	install -m 644 doc/dfrotz.6 "$(DESTDIR)$(MANDIR)/man6/"
+	install -d $(BINDIR) $(MAN_PREFIX)/man/man6
+	install -c -m 755 $(DFROTZ_BIN) $(BINDIR)
+	install -m 644 doc/dfrotz.6 $(MAN_PREFIX)/man/man6/
 
 uninstall_dumb: uninstall_dfrotz
 uninstall_dfrotz:
-	rm -f "$(DESTDIR)$(PREFIX)/bin/dfrotz"
-	rm -f "$(DESTDIR)$(MANDIR)/man6/dfrotz.6"
+	rm -f $(BINDIR)/dfrotz
+	rm -f $(MAN_PREFIX)/man/man6/dfrotz.6
 
 install_sdl: install_sfrotz
 install_sfrotz: $(SFROTZ_BIN)
-	install -d "$(DESTDIR)$(PREFIX)/bin" "$(DESTDIR)$(MANDIR)/man6"
-	install "$(SFROTZ_BIN)" "$(DESTDIR)$(PREFIX)/bin/"
-	install -m 644 doc/sfrotz.6 "$(DESTDIR)$(MANDIR)/man6/"
+	install -d $(BINDIR) $(MAN_PREFIX)/man/man6
+	install $(SFROTZ_BIN) $(BINDIR)
+	install -m 644 doc/sfrotz.6 $(MAN_PREFIX)/man/man6/
 
 uninstall_sdl: uninstall_sfrotz
 uninstall_sfrotz:
-	rm -f "$(DESTDIR)$(PREFIX)/bin/sfrotz"
-	rm -f "$(DESTDIR)$(MANDIR)/man6/sfrotz.6"
+	rm -f $(BINDIR)/sfrotz
+	rm -f $(MAN_PREFIX)/man/man6/sfrotz.6"
 
 install_all:	install_frotz install_dfrotz install_sfrotz
 
